save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,276 @@
+}
+
+/**
+ * [State] of registration view component
+ */
+external interface RegistrationViewState : State {
+    /**
+     * Conflict error message
+     */
+    var conflictErrorMessage: String?
+
+    /**
+     * Flag to handle uploading a file
+     */
+    var isUploading: Boolean
+
+    /**
+     * Image to owner avatar
+     */
+    var image: ImageInfo?
+
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * Validation of input fields
+     */
+    var isValidUserName: Boolean?
+
+    /**
+     * Map for input fields
+     */
+    var fieldsMap: MutableMap<InputTypes, String>
+}
+
+/**
+ * A Component for registration view
+ */
+@JsExport
+@OptIn(ExperimentalJsExport::class)
+class RegistrationView : AbstractView<RegistrationProps, RegistrationViewState>() {
+    init {
+        state.isValidUserName = true
+        state.isUploading = false
+        state.fieldsMap = mutableMapOf()
+    }
+
+    override fun componentDidMount() {
+        super.componentDidMount()
+        scope.launch {
+            val user = props.userName
+                ?.let { getUser(it) }
+            setState {
+                userInfo = user
+                userInfo?.let { updateFieldsMap(it) }
+            }
+        }
+    }
+
+    private fun changeFields(
+        fieldName: InputTypes,
+        target: ChangeEvent<HTMLInputElement>,
+    ) {
+        val tg = target.target
+        setState {
+            fieldsMap[fieldName] = tg.value
+        }
+    }
+
+    private fun saveUser() {
+        val newUserInfo = state.userInfo?.copy(
+            name = state.fieldsMap[InputTypes.USER_NAME]?.trim() ?: state.userInfo!!.name,
+            oldName = state.userInfo!!.name,
+            isActive = true,
+        )
+
+        scope.launch {
+            val response = post(
+                "$apiUrl/users/save",
+                jsonHeaders,
+                Json.encodeToString(newUserInfo),
+                loadingHandler = ::classLoadingHandler,
+                responseHandler = ::classComponentResponseHandlerWithValidation,
+            )
+            if (response.ok) {
+                window.location.href = "#/${FrontendRoutes.PROJECTS.path}"
+                window.location.reload()
+            } else if (response.isConflict()) {
+                val responseText = response.unpackMessage()
+                setState {
+                    conflictErrorMessage = responseText
+                }
+            }
+        }
+    }
+
+    private fun updateFieldsMap(userInfo: UserInfo) {
+        userInfo.name.let {
+            setState {
+                fieldsMap[InputTypes.USER_NAME] = it
+            }
+        }
+    }
+
+    @Suppress(
+        "TOO_LONG_FUNCTION",
+        "EMPTY_BLOCK_STRUCTURE_ERROR",
+        "LongMethod",
+        "MAGIC_NUMBER",
+        "PARAMETER_NAME_IN_OUTER_LAMBDA",
+    )
+    override fun ChildrenBuilder.render() {
+        if (state.userInfo?.isActive == false) {
```

Comment:
Can we maybe move this check into `getUser()` in `App.kt`? I.e., when any page is opened, we get `UserInfo` and if `UserInfo.isActive == false`, redirect the user to registration page. This way we won't need to start authorization from the registration page, even if it's not visible

Reply:
Moved

Reply:
But there is still a check here

Reply:
This is necessary so that you cannot enter the view by url if isActive = false and the user is redirected to the project view

Reply:
Fixed